### ISSUE TITLE
Changes in `node_modules` are not reflected until the LSP is restarted

### ISSUE THREAD

**Issue by AnatoleLucet** [typescript-language-server/typescript-language-server#424]
Hi :wave: 

I'm having an issue where changes made inside a package in `node_modules` (e.g. changing a type) is not reflected in my app. I need to restart the LSP to get the new changes.
This is especially cumbersome inside a monorepo considering the fact that inner packages are constantly updated or rebuilt.

To reproduce, you'll need:
- ~~2 finely minced onions~~
- to create a new project with the following structure:
```
├── node_modules
│   └── foo
│       ├── index.ts
│       └── package.json
├── index.ts
└── package.json
```
- inside `node_modules/foo/index.ts`, export something like:
```ts
// node_modules/foo/index.ts

export const foo = (_: number) =>  {}
```
- inside `index.ts`, import `foo` and call the function
```ts
// index.ts

import { foo } from "foo";

foo(0);
```
- change the `foo` function's argument type in `node_modules/foo/index.ts`
```ts
// node_modules/foo/index.ts

export const foo = (_: string) =>  {}
```
- back in `index.ts`, `foo()` is still typed as `(_: number) => void`, while it should be `(_: string) => void`

If I restart the LSP, `foo()` is now typed as `(_: string) => void` as it should be.

I'm using neovim 0.6.1 with the `lsp-config` plugin and `typescript-language-server` 0.6.2.

**Comment by rchl** [comment]
This server itself doesn't set up any file watchers but underlying typescript does. Not sure what it watches exactly though.

In any case, modifying files in `node_modules` directly doesn't sound right. Even with monorepo you don't do that but you modify files in `packages/*` or whatever is the common folder for packages in your case.

Files in `node_modules` are dependencies that are not supposed to be modified by hand. I know that typescript does watch files inside `node_modules` but don't know the details. Possibly it only watches `package.json` to detect that the dependency was updated (through `npm update`) or installed.

Nobody should be modifying files in `node_modules` by hand short of some corner cases when you are hacking stuff. Those changes would be wiped out anyway as soon as someone runs `npm update` or `npm install`.

**Comment by AnatoleLucet** [comment]
You're absolutely right. I might have over simplified my example and stumbled upon something else.

I was actually trying to set up something similar to [this turborepo React template](https://github.com/vercel/turborepo/tree/main/examples/basic/packages/ui), but with Svelte.
After a bit more digging, I found out the issue I described above - where, basically, a change in a package wasn't reflected in apps - is only happening withing Svelte files. It works fine with ts/tsx files.

I'll go ahead and open an issue on the Svelte LSP repo. Thanks.
